rpmio/rpmkeyring.cc

```diff
@@ -363,52 +372,59 @@ pgpDigParams rpmPubkeyPgpDigParams(rpmPubkey key)
     return params;
 }
 
-static rpmPubkey findbySig(rpmKeyring keyring, pgpDigParams sig)
+rpmRC rpmKeyringVerifySig2(rpmKeyring keyring, pgpDigParams sig, DIGEST_CTX ctx, rpmPubkey * keyptr)
 {
+    rpmRC rc = RPMRC_FAIL;
     rpmPubkey key = NULL;
+    std::vector<std::pair<int, std::string>> results = {};
 
-    if (keyring && sig) {
+    if (sig && ctx) {
+	char *lints = NULL;
```

Comment:
This will be moot once you add the wrapper around pgpVerifySignature2() which makes lints local, but this is another example of a variable in the wrong scope and in this case, dangerous: lints can be free'd and then accessed again later, but the pointer is not NULL'ed which is akin to leaving a booby trap in the code. Does pgpVerifySignature() *always* assign to the lints pointer when passed? Maybe it does, but if not then we can end up accessing freed memory. You don't want to rely on such a thing.